processing/src/main/java/org/apache/carbondata/processing/loading/sort/impl/UnsafeParallelReadMergeSorterImpl.java

```diff
@@ -85,7 +85,9 @@ public UnsafeParallelReadMergeSorterImpl(AtomicLong rowCounter) {
     } catch (Exception e) {
       throw new CarbonDataLoadingException(e);
     }
-    this.executorService = Executors.newFixedThreadPool(iterators.length,
+    // setting the number of cores to 1 in case of range sort
+    int numberOfCores = sortParameters.isRangeSort() ? 1 : iterators.length;
```

Comment:
I think for range sort iterators length is always one so no need have this condition

Reply:
Okay i will check and update

Reply:
updated code removed isRangeSort from sort parameter as sort iterators length was always one